docs/getting-started-guide.md

```diff
@@ -0,0 +1,168 @@
+# How to set up and run the Co-Simulation Demo
+
+This demonstration shows how to compile and run the Co-Simulation demo of Buildroot in QEMU with a simulated device in SystemC. This configeration is tested working for Ubuntu 18.0.4 and assumes that `cosim` is located in your home directory. This walkthough also assumes the device being emulated is the Zynq-7000 SoC.
+
+## Dependencies
+Below are the dependencies needed to compile all the libraries in this demo:
+```bash
+sudo apt update
+sudo apt install cmake gmake gcc qemu-kvm qemu-system qemu-user-static verilator
+```
+
+## Setup and Compilation
+Run these commands to clone and build the necessary repos (`~/cosim` assumed as the base directory).
+
+### Create the base directory
+```bash
+mkdir ~/cosim
+```
+
+### SystemC Setup
+```bash
+cd ~/cosim
+SYSC_VERSION=systemc-2.3.2
+wget https://www.accellera.org/images/downloads/standards/systemc/systemc-2.3.2.tar.gz
+tar xf ${SYSC_VERSION}.tar.gz && cd ${SYSC_VERSION}/
+
+# Build and install the binaries
+CXXFLAGS=-std=c++11 ./configure ; make
+sudo make install
+```
+
+### QEMU Setup
+```bash
+cd ~/cosim
+git clone https://github.com/Xilinx/qemu.git
+cd qemu
+git checkout 7e3e3ae09af8abe70383a5
+git submodule update --init dtc
+
+# Configure and build
+./configure --target-list="arm-softmmu,aarch64-softmmu,microblazeel-softmmu" --enable-fdt --disable-kvm --disable-xen
+make
+sudo make install
+export SYSTEM=/cosim/systemc-2.3.2/
+```
+
+### Demo Setup
+```bash
+# Clone the repo and the submodules
+cd ~/cosim
+git clone https://github.com/Xilinx/systemctlm-cosim-demo.git
+cd systemctlm-cosim-demo
+sed -i 's~../libsystemctlm-soc.git~https://github.com/Xilinx/libsystemctlm-soc.git~g' .gitmodules
+git submodule sync --recursive && git submodule update --init libsystemctlm-soc
+
+# Create the Makefile configeration
+cat << EOF | tee .config.mk
+CXXFLAGS=-std=c++11
+SYSTEMC_INCLUDE=${HOME}/cosim/systemc-2.3.2/src
+SYSTEMC_LIBDIR=${HOME}/cosim/systemc-2.3.2/src/.libs
```

Comment:
If the `--prefix=/usr/local/systemc-2.3.2/` is used when configuring the systemc-2.3.2  build we should be able to remove above two lines. The Makefile will pick up above as default location through this portion:

```
...
SYSTEMC ?= /usr/local/systemc-2.3.2/
SYSTEMC_INCLUDE ?=$(SYSTEMC)/include/
SYSTEMC_LIBDIR ?= $(SYSTEMC)/lib-linux64
...
```

Reply:
Installing the systemc libs in ${HOME}/cosim is probably safer as an example since the user may not have access to /usr/local/. But I think we need to include --prefix=${HOME}/cosim/ in the command-line to configure.

Reply:
@edgarigl I agree, a home install would likely make this walkthrough more accessible.